bosh-director/lib/bosh/director/jobs/update_release.rb

```diff
@@ -484,25 +485,43 @@ def create_package(package_meta, release_dir)
 
       # @return [boolean] true if a new blob was created; false otherwise
       def save_package_source_blob(package, package_meta, release_dir)
-        return false unless package.blobstore_id.nil?
-
-        name, version, existing_blob = package_meta['name'], package_meta['version'], package_meta['blobstore_id']
+        name, version, existing_blob, sha1 = package_meta['name'], package_meta['version'], package_meta['blobstore_id'], package_meta['sha1']
         desc = "package '#{name}/#{version}'"
+        package_tgz = File.join(release_dir, 'packages', "#{name}.tgz")
+
+        if @fix
+          if package.blobstore_id != nil
+            logger.info("Verifying package #{desc} with blobstore_id: #{package.blobstore_id}")
+            return false if BlobUtil.verify_blob(package.blobstore_id, package.sha1)
```

Comment:
@mariash, if we just upload the package, might introduce other issue, e.g. for the healthy package in the blobstore, if any exception happens during the uploading of the package, this may cause we use a damaged package to replace the original healthy one. and also, if we just upload the package every time, we need to update the db record and clean the old package file in the blobstore, which is also a more 'longer' process to take.